ark/segmentation/marker_quantification.py

```diff
@@ -5,16 +5,156 @@
+            morphology features for regionprops to extract for each cell
+        regionprops_extras (list):
+            list of extra properties derived from regionprops to compute
+        **kwargs:
+            Arbitrary keyword arguments for compute_extra_props
+
+    Returns:
+        pandas.DataFrame:
+            Contains the regionprops info (base and derived) for each labeled cell
+    """
+
+    cell_props = pd.DataFrame(regionprops_table(segmentation_labels,
+                                                properties=regionprops_features))
+
+    # compute the extras properties for each cell, and append to cell_props
+    props = regionprops(segmentation_labels)
+    extra_prop_data = compute_extra_props(props, regionprops_extras, **kwargs)
+    cell_props = pd.concat([cell_props, extra_prop_data], axis=1)
+
+    return cell_props
+
+
+def assign_cell_features(marker_counts, compartment, cell_props, cell_coords, cell_id,
+                         label_id, input_images, regionprops_names, extraction, **kwargs):
+    """Assign the regionprops features and signal intensity to cell_id in marker_counts
+
+    Args:
+        marker_counts (xarray.DataArray):
+            xarray containing segmentaed data of cells x markers
+        compartment (str):
+            either 'whole_cell' or 'nuclear'
+        cell_props (pandas.DataFrame):
+            regionprops information for each cell
+        cell_coords (numpy.ndarray):
+            values representing pixels within one cell
+        cell_id (int):
+            id of the cell
+        label_id (int):
+            id used to index into cell_props
+        input_images (xarray.DataArray):
+            rows x columns x channels matrix of imaging data
+        regionprops_names (list):
+            all of the regionprops features (including derived, except nuclear-specific)
+        extraction (str):
+            the extraction method to use for signal intensity calculation
+        **kwargs:
+            arbitrary keyword arguments
+    """
+
+    # get centroid corresponding to current cell
+    kwargs['centroid'] = np.array((
+        cell_props.loc[cell_props['label'] == label_id, 'centroid-0'].values,
+        cell_props.loc[cell_props['label'] == label_id, 'centroid-1'].values
+    )).T
+
+    # calculate the total signal intensity within cell
+    cell_counts = EXTRACTION_FUNCTION[extraction](cell_coords, input_images, **kwargs)
+
+    # get morphology metrics
+    current_cell_props = cell_props.loc[cell_props['label'] == label_id, regionprops_names]
+
+    # combine marker counts and morphology metrics together
+    cell_features = np.concatenate((cell_counts, current_cell_props), axis=None)
+
+    # add counts of each marker to appropriate column
+    marker_counts.loc[compartment, cell_id, marker_counts.features[1]:] = cell_features
+
+    # add cell size to first column
+    marker_counts.loc[compartment, cell_id, marker_counts.features[0]] = cell_coords.shape[0]
```

Comment:
I think for clarity of what's happening, it's better to return the updated data_frame instead of just modifying in place, makes the top-level functions easier to follow.